### PR TITLE
World Icons

### PR DESCRIPTION
**Pull request by Sun-Soaked**
## About The Pull Request
Ports world icons from Mojave 13.
This is an element that allows for tiny, size-accurate versions of object sprites in a roundabout way.

~~Drafted since spritainers want more coverage before merging~~
UNDRAFTED by request of a spritainer. God weeps

### //MAKING A SUPER BASIC WORLD ICON IMPLEMENTATION GUIDE: FOR REAL. NO VIRUS. 100% ORGANIC//

- Place the Element in an Init below The Item In Question

`/obj/item/Initialize()`
`	. = ..()`
`	AddElement(/datum/element/world_icon, null, icon, 'icon')`

- Fill in info. Put the ORIGINAL ICON in the addelement's `'icon'` slot
(yeah, this element layers the "original" sprite on top of the worldicon one, using it as the default. It just be like that sometimes)

`/obj/item/kitchen/knife/combat/Initialize()`
`	. = ..()`
`	AddElement(/datum/element/world_icon, null, icon, 'icons/obj/kitchen.dmi')`
	
- Put the directory to your NEW, WORLD ICON in the item's normal icon var
	`icon = 'icons/obj/world/melee.dmi'`	

The world icon and icon adjustment vars on obj/item should allow for implementation in places where an object's Icon has to frequently change. 

## Why It's Good For The Game
Just a little guy

## Changelog

:cl:
add: World icon element.
add: World icons for combat knives.
/:cl:

### CODE REVIEW

**Comment by Sun-Soaked** [comment]
fuck you linters

**Comment by thgvr** [comment]
: D

**Comment by Sun-Soaked** [comment]
> : D

I mean it works!
The stuff that we went over in dev is uh

-Spritainers want more sprites done before it's formally implemented in

-Tmt had some issues with the implementation, which is a little odd because it comes from a server trying to have code parity with /TG/
IMO it's better to Have It and then rework it later than wait, but it's their call

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -508,12 +512,13 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 	SHOULD_CALL_PARENT(1)
 	visual_equipped(user, slot, initial)
-	SEND_SIGNAL(src, COMSIG_ITEM_EQUIPPED, user, slot)
 	for(var/X in actions)
 		var/datum/action/A = X
 		if(item_action_slot_check(slot, user)) //some items only give their actions buttons when in a specific slot.
 			A.Grant(user)
 	item_flags |= IN_INVENTORY
+	SEND_SIGNAL(src, COMSIG_ITEM_EQUIPPED, user, slot)
+	SEND_SIGNAL(user, COMSIG_MOB_EQUIPPED_ITEM, src, slot)
```
Comment (tmtmtl30) [inline code/game/objects/items.dm]: bit of an odd addition seeing it's not used, but w/e

**Comment by tmtmtl30** [comment]
sorry for not looking at this in very long. the reason i have issues with the implementation is that it looks like it's implemented to minimize the # of lines you have to modify in an item's original file to add world sprites (why make it an element attached to the item instead of just base behavior on /obj/item?). that makes sense for a server that's trying to maintain parity with an upstream, but less so for us.

regardless, so long as it's functional (and doesn't break when things get blood on them), i'm fine with it

**Comment by Sun-Soaked** [comment]
ouaagh(mojave  why.. .)

**Comment by thgvr** [comment]
Any luck with the overlay stuff?

File: code/game/objects/items.dm (hunk 1 above)
Comment (Sun-Soaked) [reply code/game/objects/items.dm]: oh this was actually a misunderstanding. whoopsies!

**Comment by Sun-Soaked** [comment]
overlays now working, but at what cost? (don't look at the bottom of blood.dm)

**Comment by FalloutFalcon** [comment]
Pretty sure this just caused a bunch of items to randomly have no lighting and be on a really high sprite layer

**Pull request merged**